src/api/wiki.js

```diff
@@ -19,30 +19,51 @@ const getArticle = (lang, title) => {
   const url = `https://${lang}.wikipedia.org/api/rest_v1/page/mobile-sections/${title}`
   return cachedFetch(url, data => {
     const parser = new DOMParser()
-    let content = data.lead.sections[0].text
+    const title = data.lead.displaytitle
+    const imageUrl = data.lead.image && data.lead.image.urls['320']
+
+    // parse info box
     const doc = parser.parseFromString(data.lead.sections[0].text, 'text/html')
     const infoboxNode = doc.querySelector('[class^="infobox"]')
     const infobox = infoboxNode && infoboxNode.outerHTML
 
+    // parse lead as the first section
+    const sections = []
+    sections.push({
+      description: data.lead.description,
+      content: data.lead.sections[0].text
+    })
+
+    // parse section as the remaining section
+    let nextDescription = ''
+    let nextContent = ''
     data.remaining.sections.forEach((s) => {
+      // the section starts with every toclevel 1
+      if (s.toclevel === 1 && nextDescription) {
+        sections.push({
+          description: nextDescription, content: nextContent
+        })
+
+        nextDescription = ''
+        nextContent = ''
+      }
+
+      if (s.toclevel === 1) nextDescription = s.line
+
       const header = 'h' + (s.toclevel + 1)
       const headerLine = `<${header}>${s.line}</${header}>`
-      content += headerLine
-      content += s.text
-    })
 
-    // The app is served from the app:// protocol so protocol-relative
-    // image sources don't work.
-    content = content.replace(/src="\/\//gi, 'src="https://')
+      // The app is served from the app:// protocol so protocol-relative
+      // image sources don't work.
+      nextContent += headerLine + s.text.replace(/src="\/\//gi, 'src="https://')
```

Comment:
The header should not be part of the content since it will be show as the title on the title page.

Reply:
currently the section starts with `toclevel 1`, `toclevel 2` and the following will grouped in the previous `toclevel 1`, 

I will change the current behaviour to include the headerLine only when the toclevel is not 1, let me know if this is fine.